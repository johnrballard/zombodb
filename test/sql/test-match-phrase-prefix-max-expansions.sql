select count(*) from so_posts where so_posts ==> 'body: "java is*"';
select count(*) from so_posts where so_posts ==> '({
 "match_phrase_prefix": {
   "body": {
     "query": "java is",
     "boost": 1.0,
     "max_expansions": 2147483647
   }
 }
})';
select zdb.dump_query('so_posts', 'body: "java is*"');

select count(*) from so_posts where so_posts ==> 'body: "java i*"';
select count(*) from so_posts where so_posts ==> '({
 "match_phrase_prefix": {
   "body": {
     "query": "java i",
     "boost": 1.0,
     "max_expansions": 2147483647
   }
 }
})';
select zdb.dump_query('so_posts', 'body: "java i*"');