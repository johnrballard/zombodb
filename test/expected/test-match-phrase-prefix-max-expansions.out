select count(*) from so_posts where so_posts ==> 'body: "java is*"';
 count 
-------
   859
(1 row)

select count(*) from so_posts where so_posts ==> '({
 "match_phrase_prefix": {
   "body": {
     "query": "java is",
     "boost": 1.0,
     "max_expansions": 2147483647
   }
 }
})';
 count 
-------
   859
(1 row)

select zdb.dump_query('so_posts', 'body: "java is*"');
             dump_query             
------------------------------------
 {                                 +
   "match_phrase_prefix": {        +
     "body": {                     +
       "query": "java is",         +
       "boost": 1.0,               +
       "max_expansions": 2147483647+
     }                             +
   }                               +
 }
(1 row)

select count(*) from so_posts where so_posts ==> 'body: "java i*"';
 count 
-------
  2075
(1 row)

select count(*) from so_posts where so_posts ==> '({
 "match_phrase_prefix": {
   "body": {
     "query": "java i",
     "boost": 1.0,
     "max_expansions": 2147483647
   }
 }
})';
 count 
-------
  2075
(1 row)

select zdb.dump_query('so_posts', 'body: "java i*"');
             dump_query             
------------------------------------
 {                                 +
   "match_phrase_prefix": {        +
     "body": {                     +
       "query": "java i",          +
       "boost": 1.0,               +
       "max_expansions": 2147483647+
     }                             +
   }                               +
 }
(1 row)

